src/3d/qgspointcloudlayer3drenderer.cpp

```diff
@@ -241,3 +251,42 @@ bool QgsPointCloudLayer3DRenderer::convertFrom2DRenderer( QgsPointCloudRenderer
   setSymbol( newSymbol );
   return true;
 }
+
+bool QgsPointCloudLayer3DRenderer::supportsUpdateFrom( QgsAbstract3DRenderer *renderer ) const
+{
+  QgsPointCloudLayer3DRenderer *r = dynamic_cast<QgsPointCloudLayer3DRenderer *>( renderer );
+  if ( !r || !r->symbol() )
+  {
+    return false;
+  }
+
+  const QgsPointCloud3DSymbol *rSymbol = r->symbol();
+  if ( rSymbol->symbolType() != mSymbol->symbolType() )
+  {
+    return false;
+  }
+
+  return true;
+}
+
+bool QgsPointCloudLayer3DRenderer::updateCurrentRenderer( QgsAbstract3DRenderer *renderer )
+{
+  Q_ASSERT( supportsUpdateFrom( renderer ) );
+  QgsPointCloudLayer3DRenderer *r = dynamic_cast<QgsPointCloudLayer3DRenderer *>( renderer );
+  setShowBoundingBoxes( r->showBoundingBoxes() );
+  bool updateScene = false;
+  if ( r->maximumScreenError() != maximumScreenError() )
+  {
+    setMaximumScreenError( r->maximumScreenError() );
+    updateScene = true;
+  }
+  if ( r->pointRenderingBudget() != pointRenderingBudget() )
+  {
+    setPointRenderingBudget( r->pointRenderingBudget() );
+    updateScene = true;
+  }
+  // TODO: fix const access
+  updateScene = updateScene || mSymbol->updateCurrentSymbol( const_cast< QgsPointCloud3DSymbol * >( r->symbol() ) );
+  return updateScene;
+}
```

Comment:
Why did not you merge theses 2 functions? One should be enough?

```c++
bool QgsPointCloudLayer3DRenderer::updateCurrentRenderer( QgsAbstract3DRenderer *renderer )
{
  QgsPointCloudLayer3DRenderer *r = dynamic_cast<QgsPointCloudLayer3DRenderer *>( renderer );
  if ( !r || !r->symbol() )
  {
    return false;
  }

  const QgsPointCloud3DSymbol *rSymbol = r->symbol();
  if ( rSymbol->symbolType() != mSymbol->symbolType() )
  {
    return false;
  }
  if ( r->maximumScreenError() == maximumScreenError() || r->pointRenderingBudget() == pointRenderingBudget())
  {
    return false;
  }

  setShowBoundingBoxes( r->showBoundingBoxes() );
  setMaximumScreenError( r->maximumScreenError() );
  setPointRenderingBudget( r->pointRenderingBudget() );
  // TODO: fix const access
  updateScene = updateScene || mSymbol->updateCurrentSymbol( const_cast< QgsPointCloud3DSymbol * >( r->symbol() ) );
  return true;
}
```

Reply:
For me it seems simpler to first check for updating support than to handle 3 the following cases in one go:
- The renderer is of same symbol as the new renderer but could be updated without resetting 
- The renderer is of same symbol as the new renderer but should be updated without resetting 
- The renderer is of a different symbol and needs to be completely reset